app/models/planting.rb

```diff
@@ -8,6 +8,8 @@ class Planting < ActiveRecord::Base
   belongs_to :crop, counter_cache: true
   has_many :harvests, -> { order(harvested_at: :desc) }, dependent: :destroy
 
+  before_save :calc_and_set_days_before_maturity
```

Comment:
So, in the controller's `update()` and `create()`, we'd call `@planting.calc_and_set_days_before_maturity()` before save?